src/main/kotlin/org/rust/ide/intentions/DemorgansLawIntention.kt

```diff
@@ -71,18 +71,30 @@ class DemorgansLawIntention : RsElementBaseIntentionAction<DemorgansLawIntention
     private fun applyDemorgan(project: Project, topBinExpr: RsBinaryExpr, opType: BinaryOperator) {
         val converted = convertConjunctionExpression(topBinExpr, opType) ?: return
 
-        var expressionToReplace: RsExpr = topBinExpr
-        var expString = "!($converted)"
         val parent = topBinExpr.parent.parent
```

Comment:
Maybe `parent` should be checked for `null` value. Not sure if it's a common situation but it's possible at least when `topBinExpr` is a child of `RsReplCodeFragment` or something like that